### ISSUE TITLE
[CST] Review accessibility of claim letter PDFs

### ISSUE THREAD

**Issue by davidakennedy** [department-of-veterans-affairs/va.gov-team#95017]
## Value Statement

**_As a_** Veteran
**_I want to_** access accessible claim letter PDFs
**_So that_** I can read them and store them how I want to on my own device

---
### New Feature

We suspect that claim letter PDFs aren't accessible, but we don't exactly what's wrong with the PDFs. In order to potentially deliver accessible PDFs, we need to understand what's making them inaccessible. According to @jstrothman - "While our team doesn't generate the letters, we do deliver them, so I think we should at the very least know what, if any, accessibility issues exist." For more, [check out this Slack thread](https://dsva.slack.com/archives/C06V3RU6HA9/p1725993257419269).

## Outcome, Success Measure, KPI(S), and Tracking Link
- Add here

## Design

From an accessibility perspective, we will:

- Test at one type of each letter via Adobe Acrobat Pro, and via screen reader, and document accessibility problems.

List of letters:

- 184 (doc type) - Decision Notices
- 34 - Correspondence
- 408 - VA Examination Letter
- 700 - MAP-D Development Letter
- 859 - Subsequent Development Letter
- 940 - Final Attempt Letter
- 864 - General Records Request (Medical)
- 1605 - General Records Request (Non-Medical)
- 704 - 5103
- 706 - 5103
- 858 - 5103

### Enablement team (if needed)

@jstrothman, for awareness.

## Engineering 

N/A.

## Out of scope

- A lot of the possible fixes may be out of our team's control, and would need to be shared with other teams.

## Open questions

- Here, I want to outline what teams control the PDFs, so I can talk to them, if nevessary.

## Tasks
- [ ] Audit roughly nine letters.
- [ ] Write recommendations for improvements.

## Definition of Done
- [ ] Meets acceptance criteria
- [ ] Passed E2E testing (90% coverage)
- [ ] Passed unit testing (90% coverage)
- [ ] Passed integration testing (if applicable)
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Team approved production verification process
- [ ] Design performs design QA and verifies the implementation matches the design spec
- [ ] Accessibility specialist performs accessibility review (in code or design)
- [ ] Engineering identifies staging users required to test and shares account and credentials with design and product
- [ ] Product performs functional QA and verifies acceptance criteria was met

## Acceptance Criteria
- [ ] Mini audit created.
- [ ] Suggest recommendations for improvement

**Comment by davidakennedy** [comment]
I've done some testing with some of our existing PDF samples. Here's what I found so far.

This is the list I worked from:

- 184 - Decision Notices
- 34 - Correspondence
- 408 - VA Examination Letter
- 700 - MAP-D Development Letter
- 859 - Subsequent Development Letter
- 940 - Final Attempt Letter
- 864 - General Records Request (Medical)
- 1605 - General Records Request (Non-Medical)
- 704 - 5103
- 706 - 5103
- 858 - 5103

The first number there is the document ID for the claim letter.

What I've tested so far:

| File name                                                                  | Letter type                                        | Accessibility issues                                                                                                                                                                                                                                                                              | Does it read in a screen reader? |
| -------------------------------------------------------------------------- | -------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------- |
| ClaimLetter-3p-UAT VBMS - general medical records request - 864.pdf        | 864 - (doc type) General Records Request (Medical) | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |
| ClaimLetter-2022-4-12 staging - disability comp - service connex found.pdf | 184 - (doc type) Decision Notices                  | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |
| ClaimLetter-2024-10-15-704.pdf                                             | 704 (doc type) - 5103                              | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |
| ClaimLetter-2016-5-20-706.pdf                                              | 706 (doc type) - 5103                              | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |
| ClaimLetter-2024-8-12-858.pdf                                              | 858 (doc type) - 5103                              | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |

Some more context on that testing:

- I've tested five out of the 11 types of PDFs. Andrew Gray is working on getting me more PDF samples, [per this thread in Slack](https://dsva.slack.com/archives/C04KHCT3ZMY/p1730743832813609?thread_ts=1730235741.408839&cid=C04KHCT3ZMY), to test. I should be able to get four out of the five remaining ones. Missing one PDF shouldn't make a meaningful difference in test coverage because so far the PDFs have tested nearly identically.
- All of the PDFs I tested had identical or very similar results as far as issues. There isn't wide variation in the issues, which leads me to believe that all the PDFs are created in the same way.

Lastly, I also got more information from Andrew regarding how the PDFs are made. I'll be placing that in #96066.

**Comment by davidakennedy** [comment]
I've completed testing with eight out the 11 PDF samples. After [talking with Julie](https://dsva.slack.com/archives/C06V3RU6HA9/p1731966429381049?thread_ts=1729110371.365479&cid=C06V3RU6HA9), we decided this was enough of a sample. Here's what I found so far.

This is the list I worked from:

- 184 - Decision Notices
- 34 - Correspondence
- 408 - VA Examination Letter
- 700 - MAP-D Development Letter
- 859 - Subsequent Development Letter
- 940 - Final Attempt Letter
- 864 - General Records Request (Medical)
- 1605 - General Records Request (Non-Medical)
- 704 - 5103
- 706 - 5103
- 858 - 5103

The samples not tested:

- 34 - Correspondence
- 700 - MAP-D Development Letter
- 1605 - General Records Request (Non-Medical)

## What was tested

| File name                                                                  | Letter type                                        | Accessibility issues                                                                                                                                                                                                                                                                              | Does it read in a screen reader? |
| -------------------------------------------------------------------------- | -------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------- |
| ClaimLetter-3p-UAT VBMS - general medical records request - 864.pdf        | 864 - (doc type) General Records Request (Medical) | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |
| ClaimLetter-2022-4-12 staging - disability comp - service connex found.pdf | 184 - (doc type) Decision Notices                  | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |
| ClaimLetter-2024-10-15-704.pdf                                             | 704 (doc type) - 5103                              | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |
| ClaimLetter-2016-5-20-706.pdf                                              | 706 (doc type) - 5103                              | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |
| ClaimLetter-2024-8-12-858.pdf                                              | 858 (doc type) - 5103                              | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |
| LT3123 - ExamAppointmentLetter - 11122024-408.pdf                          | 408 (doc type) - VA Examination Letter             | PDF doesn't have a title; PDF lacks alternative text in spots; Tab order issues in spots                                                                                                                                                                                                          | Yes                              |
| LT3123 - FinalAttemptLetterStrs - 11122024-940.pdf                         | 940 (doc type) - Final Attempt Letter              | PDF doesn't have a title                                                                                                                                                                                                                                                                          | Yes                              |
| Subsequent Development Letter.pdf                                          | 859 (doc type) - Subsequent Development Letter     | PDF doesn't contain tags; PDF doesn't contain primary language; PDF doesn't have a title; PDF has a character encoding error(s); PDF lacks alternative text in spots; Tables in PDF have issues with rows/columns/headers, etc.; List items are structured properly; Headings lack proper nesting | Yes                              |

Some more context on that testing:

- Andrew Gray couldn't get the "700 - MAP-D Development Letter" sample. The other two missing shouldn't make a big difference in testing.
- Six out of the eight PDFs have near identical accessibility issues, and lots of them. The other two have fewer accessibility issues, which are more minor.

I'm going to dig into more about how these are created in #96066.

## How to fix the issues

- The software that makes the PDFs (Adobe Document Generation) needs to be able to add [tags to PDFs](https://webaim.org/techniques/acrobat/#pdftags).
- The software that makes the PDFs (Adobe Document Generation) needs to be able to set a primary language.
- The software that makes the PDFs (Adobe Document Generation) needs to be able to set alternative text, including marking them decorative.
- In general, the ability to properly tag content would solve not just the tagging issues, but the alt text, problems with tables, list items and headings.
